models/OrderModel.php

```diff
@@ -1689,21 +1689,60 @@ public static function is_manual_payment( $method_name ) {
 	 * @return void
 	 */
 	public static function render_pay_button( $order ) {
+
 		if ( is_numeric( $order ) ) {
 			$order = ( new self() )->get_order_by_id( $order );
 		}
 
-		if ( self::should_show_pay_btn( $order ) ) {
-			?>
+		if ( ! self::check_expiry_time( $order ) ) { ?>
+
+			<div class="tooltip-wrap tooltip-icon">	
+				<span class="tooltip-txt tooltip-left">
+					<?php esc_html_e( 'Payment Is Pending Due To Gateway Processing.', 'tutor' ); ?>
+				</span>
+			</div>
+		
+		<?php } elseif ( self::should_show_pay_btn( $order ) ) { ?>
+			
 			<form method="post">
 				<?php tutor_nonce_field(); ?>
 				<input type="hidden" name="tutor_action" value="tutor_pay_incomplete_order">
 				<input type="hidden" name="order_id" value="<?php echo esc_attr( $order->id ); ?>">
+				
 				<button type="submit" class="tutor-btn tutor-btn-sm tutor-btn-outline-primary">
 					<?php esc_html_e( 'Pay', 'tutor' ); ?>
-				</button>
+				</button>				
 			</form>
+			
 			<?php
 		}
 	}
+
+	/**
+	 * Checks if the Repay Order-Time expired based on stored expiry time.
+	 *
+	 * @since 3.3.0
+	 *
+	 * @param object $order The order object containing order details.
+	 *
+	 * @return bool Returns true if the order is expired or expiry time is not set, otherwise false.
+	 */
+	private static function check_expiry_time( $order ) {
+
+		$current_time = DateTimeHelper::now()->to_timestamp();
+		$meta_key     = "tutor_order_id_$order->id";
+		$meta_value   = get_user_meta( get_current_user_id(), $meta_key, false );
+		$expiry_time  = ! empty( $meta_value ) ? json_decode( $meta_value[0] )->expiry_time : null;
+
+		if ( ! is_null( $expiry_time ) && $expiry_time < $current_time ) {
```

Comment:
Instead of `! is_null( $expiry_time )`, just use `!$expiry_time`. It seems logic operand should be OR.